src/vdom/component.js

```diff
@@ -144,6 +144,9 @@ export function renderComponent(component, opts, mountAll, isChild) {
 		if (initialBase && base!==initialBase && inst!==initialChildComponent) {
 			let baseParent = initialBase.parentNode;
 			if (baseParent && base!==baseParent) {
+				if (toUnmount) {
```

Comment:
I'd really love to merge this PR but I'm a bit hesitant because of this change here, because this changes the behaviour of preact itself instead of only updating the TypeScript types. 

@arusanov Can you split this change here into its own PR?